src/tabs/tab-header-bar.tsx

```diff
@@ -299,11 +313,13 @@ export function TabHeaderBar({
       'aria-controls': `${idNamespace}-${tab.id}-panel`,
       'data-testid': tab.id,
       id: getTabElementId({ namespace: idNamespace, tabId: tab.id }),
-      children: <span className={styles['tabs-tab-label']}>{tab.label}</span>,
     };
 
     if (tab.disabled) {
       commonProps['aria-disabled'] = 'true';
+      commonProps.onClick = (event: React.MouseEvent) => {
+        event.preventDefault();
```

Comment:
why this is needed?

Reply:
Checked once again, it is not needed. Updated

Reply:
upd: it's needed when a tab with href is disabled

Reply:
let's add a code comment then to make it transparent

Reply:
made this even simpler by removing this additional handler and attaching existing handler unconditionally (regardless of being `disabled`), because it prevents default for disabled tabs anyway

https://github.com/cloudscape-design/components/blob/feat/disabled-reasons-tabs/src/tabs/tab-header-bar.tsx#L401